### ISSUE TITLE
Is sending notifications on every component update even when the status is the same

### ISSUE THREAD

**Issue by squaricdot** [CachetHQ/Cachet#3267]
Running dockerized 2.3.15 cachet.

Performing healthchecks with some python code to my services and updating cachet accordingly via the API.

when an component is updated with the same status code as already registered in Cachet, Cachet is still sending e-mail notifications when the component is "updated", though it hasn't actually changed.

This makes it unusable, because I am just bombing the subscribers with the same interval as I am checking my services.

I try to update the component without providing the status code hoping cachet is smart enough to deal with it. But, even the API documentation does not say it is a mandatory value, the component get's now updated with statuscode 0, and offcourse a notification.

It seems rather odd to me , that I would be the only one experiencing this issue.

Could somebody please tell me what I am doing wrong? Should I just not update an component when the status hasn't changed. This would leave the "last update" of the component, theoretically a year ago, if I wouldnt have downtime for a year.

**Comment by squaricdot** [comment]
https://github.com/CachetHQ/Cachet/blob/2.4/app/Bus/Handlers/Events/Component/SendComponentUpdateEmailNotificationHandler.php

```
// Don't email anything if the status hasn't changed.
       if ($event->original_status === $event->new_status) {
           return;
       }
```

doesn't seem to be doing the trick i guess.

**Comment by jbrooksuk** [comment]
This is fixed in the 2.4 branch.
